homeassistant/components/livisi/config_flow.py

```diff
@@ -47,30 +42,55 @@ async def async_step_user(
         else:
             controller_info: dict[str, Any] = {}
             with suppress(ClientConnectorError):
-                controller_info = await self.aio_livisi.async_get_controller()
+                controller_info = await aio_livisi.async_get_controller()
             if controller_info:
                 return await self.create_entity(user_input, controller_info)
-            errors["base"] = "cannot_connect"
 
         return self.async_show_form(
-            step_id="user", data_schema=self.data_schema, errors=errors
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self.reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
         )
+        return await self.async_step_reauth_confirm()
 
-    async def _login(self, user_input: dict[str, str]) -> None:
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=DATA_SCHEMA,
+            )
+        return await self.async_step_user(user_input)
```

Comment:
For styling, I would invert the logic:
```suggestion
        if user_input:
            return await self.async_step_user(user_input)
        return self.async_show_form(
            step_id="reauth_confirm",
            data_schema=DATA_SCHEMA,
        )
```

Reply:
I changed this.